docs/src/lib/indexing.md

```diff
@@ -21,77 +21,146 @@ The rules for a valid type of index into a column are the following:
     * a vector of `Symbol` (does not have to be a subtype of `AbstractVector{Symbol}`);
     * a vector of `Integer` other than `Bool` (does not have to be a subtype of `AbstractVector{<:Integer}`);
     * a vector of `Bool` that has to be a subtype of `AbstractVector{Bool}`;
-    * a colon;
-    * a regular expression, which gets expanded to a vector of matching column names.
+    * a regular expression, which gets expanded to a vector of matching column names;
+    * a `Not` expression;
+    * a colon `:`.
 
 The rules for a valid type of index into a row are the following:
 * a value, later denoted as `row`:
     * an `Integer` that is not `Bool`;
 * a vector, later denoted as `rows`:
     * a vector of `Integer` other than `Bool` (does not have to be a subtype of `AbstractVector{<:Integer}`);
     * a vector of `Bool` that has to be a subtype of `AbstractVector{Bool}`;
-    * a colon.
+    * a `Not` expression;
+* a colon `:`;
+* an exclamation mark `!`.
+
+Additionally it is allowed to index into an `AbstractDataFrame` using a two dimensional `CartesianIndex`.
 
 In the descriptions below `df` represents a `DataFrame`, `sdf` is a `SubDataFrame` and `dfr` is a `DataFrameRow`.
 
-## `getindex`
+Note that `df.col` works like `df[!, col]` when getting column from an `AbstractDataFrame` or broadcasting into `df.col`,
+but it works like `df[:, col]` when used in setting column context. This behavior was chosen
+as this kind of behavior is typically expected as a reasonable default.
+
+## `getindex` and `view`
 
-The following list specifies return types of `getindex` operations depending on argument types.
+The following list specifies return types of `getindex` and `view` operations depending on argument types.
```

Comment:
Not just return types: the behavior is fully specified. Same for `setindex!`.

Reply:
fixed